apps/live/src/redis.ts

```diff
@@ -0,0 +1,210 @@
+    if (!RedisManager.instance) {
+      RedisManager.instance = new RedisManager();
+    }
+    return RedisManager.instance;
+  }
+
+  public async initialize(): Promise<void> {
+    if (this.redisClient && this.isConnected) {
+      logger.info("Redis client already initialized and connected");
+      return;
+    }
+
+    if (this.connectionPromise) {
+      logger.info("Redis connection already in progress, waiting...");
+      await this.connectionPromise;
+      return;
+    }
+
+    this.connectionPromise = this.connect();
+    await this.connectionPromise;
+  }
+
+  private getRedisUrl(): string {
+    const redisUrl = process.env.REDIS_URL?.trim();
+    const redisHost = process.env.REDIS_HOST?.trim();
+    const redisPort = process.env.REDIS_PORT?.trim();
+
+    if (redisUrl) {
+      return redisUrl;
+    }
+
+    if (redisHost && redisPort && !Number.isNaN(Number(redisPort))) {
+      return `redis://${redisHost}:${redisPort}`;
+    }
+
+    return "";
+  }
+
+  private async connect(): Promise<void> {
+    try {
+      const redisUrl = this.getRedisUrl();
+
+      if (!redisUrl) {
+        logger.warn("No Redis URL provided, Redis functionality will be disabled");
+        this.isConnected = false;
+        return;
+      }
+
+      this.redisClient = new Redis(redisUrl, {
+        lazyConnect: true,
+        keepAlive: 30000,
+        connectTimeout: 10000,
+        commandTimeout: 5000,
+        enableOfflineQueue: false,
+        maxRetriesPerRequest: 3,
+      });
+
+      // Set up event listeners
+      this.redisClient.on("connect", () => {
+        logger.info("Redis client connected");
+        this.isConnected = true;
+      });
+
+      this.redisClient.on("ready", () => {
+        logger.info("Redis client ready");
+        this.isConnected = true;
+      });
+
+      this.redisClient.on("error", (error) => {
+        logger.error("Redis client error:", error);
+        this.isConnected = false;
+      });
+
+      this.redisClient.on("close", () => {
+        logger.warn("Redis client connection closed");
+        this.isConnected = false;
+      });
+
+      this.redisClient.on("reconnecting", () => {
+        logger.info("Redis client reconnecting...");
+        this.isConnected = false;
+      });
+
+      // Connect to Redis
+      await this.redisClient.connect();
+
+      // Test the connection
+      await this.redisClient.ping();
+      logger.info("Redis connection test successful");
+    } catch (error) {
+      logger.error("Failed to initialize Redis client:", error);
+      this.isConnected = false;
+      throw error;
+    } finally {
+      this.connectionPromise = null;
+    }
+  }
+
+  public getClient(): Redis | null {
+    if (!this.redisClient || !this.isConnected) {
+      logger.warn("Redis client not available or not connected");
+      return null;
+    }
+    return this.redisClient;
+  }
+
+  public isClientConnected(): boolean {
+    return this.isConnected && this.redisClient !== null;
```

Comment:
Can we use `this.redisClient.status === 'ready'` here? Relying on the driver to determine if we are connected should provide a more accurate answer. https://github.com/redis/ioredis/blob/bedcfb5/lib/Redis.ts#L43